cmd/prepare_worker.go

```diff
@@ -0,0 +1,161 @@
+		return nil, fmt.Errorf("directory name required")
+	} else if strings.ContainsAny(b.dirName, `/\`) {
+		return nil, fmt.Errorf("dir name is not a full path, it is a single name")
+	}
+	if b.logger == nil {
+		b.logger = b.loggingOptions.MustCreateLogger()
+	}
+	lang, err := normalizeLangName(b.language)
+	if err != nil {
+		return nil, err
+	}
+	baseDir := filepath.Join(rootDir(), "workers", lang)
+	b.logger.Infof("Building %v program at %v", lang, filepath.Join(baseDir, b.dirName))
+	// Check or create dir
+	if b.createDir {
+		if err := os.Mkdir(filepath.Join(baseDir, b.dirName), 0755); err != nil {
+			return nil, fmt.Errorf("failed creating directory: %w", err)
+		}
+	} else {
+		if _, err := os.Stat(filepath.Join(baseDir, b.dirName)); err != nil {
+			return nil, fmt.Errorf("failed finding directory: %w", err)
+		}
+	}
+	switch lang {
+	case "go":
+		return b.buildGo(ctx, baseDir)
+	case "python":
+		return b.buildPython(ctx, baseDir)
+	default:
+		return nil, fmt.Errorf("unrecognized language %v", lang)
+	}
+}
+
+func (b *workerBuilder) buildGo(ctx context.Context, baseDir string) (sdkbuild.Program, error) {
+	prog, err := sdkbuild.BuildGoProgram(ctx, sdkbuild.BuildGoProgramOptions{
+		BaseDir: baseDir,
+		DirName: b.dirName,
+		Version: b.version,
+		GoModContents: `module github.com/temporalio/omes-worker
+
+go 1.20
+
+require github.com/temporalio/omes v1.0.0
+require github.com/temporalio/omes/workers/go v1.0.0
+
+replace github.com/temporalio/omes => ../../../
+replace github.com/temporalio/omes/workers/go => ../`,
+		GoMainContents: `package main
+
+import "github.com/temporalio/omes/workers/go/worker"
+
+func main() {
+	worker.Main()
+}`,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed preparing: %w", err)
+	}
+	return prog, nil
+}
+
+func (b *workerBuilder) buildPython(ctx context.Context, baseDir string) (sdkbuild.Program, error) {
+	// If version not provided, try to read it from pyproject.toml
+	version := b.version
+	if version == "" {
+		b, err := os.ReadFile(filepath.Join(baseDir, "pyproject.toml"))
+		if err != nil {
+			return nil, fmt.Errorf("failed reading package.json: %w", err)
+		}
+		for _, line := range strings.Split(string(b), "\n") {
+			line = strings.TrimSpace(line)
+			if strings.HasPrefix(line, "temporalio = ") {
+				version = line[strings.Index(line, `"`)+1 : strings.LastIndex(line, `"`)]
+				break
+			}
+		}
+		if version == "" {
+			return nil, fmt.Errorf("version not found in pyproject.toml")
+		}
+	}
+
+	// Build
+	prog, err := sdkbuild.BuildPythonProgram(ctx, sdkbuild.BuildPythonProgramOptions{
+		BaseDir:        baseDir,
+		DirName:        b.dirName,
+		Version:        version,
+		DependencyName: "omes",
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed preparing: %w", err)
+	}
+	return prog, nil
+}
+
+func rootDir() string {
+	_, currFile, _, _ := runtime.Caller(0)
+	return filepath.Dir(filepath.Dir(currFile))
+}
+
+func normalizeLangName(lang string) (string, error) {
```

Comment:
nit: Might as well put this as a helper in sdkbuild.

Reply:
No, this is not the same implementation as features repo. Not only do we not support all langs here (yet), but we normalize here to the full name matching the worker folder whereas features repo normalizes to the file extension. They are different use cases.